src/modules/market/components/market-view/market-view.jsx

```diff
@@ -20,6 +33,21 @@ export default class MarketView extends Component {
     this.clearSelectedOutcomes = this.clearSelectedOutcomes.bind(this)
   }
 
+  componentWillMount() {
+    if (this.props.isConnected && !this.props.isMarketLoaded) {
+      this.props.loadFullMarket()
+    }
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    if (
+      (this.props.isConnected === false && nextProps.isConnected === true) ||
+      this.props.marketId !== nextProps.marketId
```

Comment:
We don’t actually need this conditional here as this won’t change w/out re-routing/re-mounting.